packages/ckeditor5-dev-tests/bin/check-dependencies.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const fs = require( 'fs' );
+const path = require( 'path' );
+const glob = require( 'glob' );
+
+const cwd = process.cwd();
+const packageJson = require( path.join( cwd, 'package.json' ) );
+
+const currentPackageName = packageJson.name;
+const dependencies = Object.keys( packageJson.dependencies || {} );
+const devDependencies = Object.keys( packageJson.devDependencies || {} );
+
+const missingDependencies = new Set();
+const missingDevDependencies = new Set();
+const invalidFiles = new Set();
+
+const globPattern = path.join( cwd, '@(src|tests)/**/*.js' );
+
+for ( const filePath of glob.sync( globPattern ) ) {
+	fs.readFileSync( filePath, 'utf-8' )
+		.split( '\n' )
+		.forEach( line => {
+			// Find required package name.
+			let packageName = /^import .* from '(@ckeditor\/[^/]+)\/.*/.exec( line );
+
+			if ( !packageName ) {
+				return;
+			}
+
+			packageName = packageName[ 1 ];
+
+			// Current package cannot be added as dependency.
+			if ( currentPackageName === packageName ) {
```

Comment:
You actually shouldn't have a non-relative import within a package itself. So this is a bug as well. I'm too lazy to look for bugs caused by this now, but there were a couple. https://github.com/ckeditor/ckeditor5/issues/437 mentions this.

Reply:
Where's this condition now?

Reply:
I removed it. If a package requires some files from itself, it should use a relative path. I fixed the paths on `t/ckeditor5-dev/278` branches in several repositories.

Reply:
So please bring it back and make the tool report them in a clear (I guess they are reported now as pkg x is missing in pkg x's deps). This is a bit different case and could be reported separately.

Reply:
https://github.com/ckeditor/ckeditor5-dev/issues/282.